tests/test_strong_branching.py

```diff
@@ -0,0 +1,201 @@
+
+        # Update the bounds of the down node and up node
+        if self.scip.allColsInLP() and not self.idempotent:
+            if down_child is not None and down_bounds[best_cand_idx] is not None:
+                self.scip.updateNodeLowerbound(down_child, down_bounds[best_cand_idx])
+            if up_child is not None and up_bounds[best_cand_idx] is not None:
+                self.scip.updateNodeLowerbound(up_child, up_bounds[best_cand_idx])
+
+        return {"result": SCIP_RESULT.BRANCHED}
+
+
+class FeatureSelectorBranchingRule(Branchrule):
+
+    def __init__(self, scip):
+        self.scip = scip
+
+    def branchexeclp(self, allowaddcons):
+
+        if self.scip.getNNodes() == 1 or self.scip.getNNodes() == 250:
+
+            rows = self.scip.getLPRowsData()
+            cols = self.scip.getLPColsData()
+
+            # This is just a dummy rule to check functionality.
+            # A user should be able to see how they can access information without affecting the solve process.
+
+            age_row = rows[0].getAge()
+            age_col = cols[0].getAge()
+            red_cost_col = self.scip.getColRedCost(cols[0])
+
+            avg_sol = cols[0].getVar().getAvgSol()
+
+        return {"result": SCIP_RESULT.DIDNOTRUN}
+
+
+def create_model():
+    scip = Model()
+    scip.setHeuristics(SCIP_PARAMSETTING.OFF)
+    scip.setSeparating(SCIP_PARAMSETTING.OFF)
+    scip.setLongintParam("limits/nodes", 2000)
+
+    x0 = scip.addVar(lb=-2, ub=4)
+    r1 = scip.addVar()
+    r2 = scip.addVar()
+    y0 = scip.addVar(lb=3)
+    t = scip.addVar(lb=None)
+    l = scip.addVar(vtype="I", lb=-9, ub=18)
+    u = scip.addVar(vtype="I", lb=-3, ub=99)
+
+    more_vars = []
+    for i in range(100):
+        more_vars.append(scip.addVar(vtype="I", lb=-12, ub=40))
+        scip.addCons(quicksum(v for v in more_vars) <= (40 - i) * quicksum(v for v in more_vars[::2]))
+
+    for i in range(100):
+        more_vars.append(scip.addVar(vtype="I", lb=-52, ub=10))
+        scip.addCons(quicksum(v for v in more_vars[50::2]) <= (40 - i) * quicksum(v for v in more_vars[405::2]))
+
+    scip.addCons(r1 >= x0)
+    scip.addCons(r2 >= -x0)
+    scip.addCons(y0 == r1 + r2)
+    scip.addCons(t + l + 7 * u <= 300)
+    scip.addCons(t >= quicksum(v for v in more_vars[::3]) - 10 * more_vars[5] + 5 * more_vars[9])
+    scip.addCons(more_vars[3] >= l + 2)
+    scip.addCons(7 <= quicksum(v for v in more_vars[::4]) - x0)
+    scip.addCons(quicksum(v for v in more_vars[::2]) + l <= quicksum(v for v in more_vars[::4]))
+
+    scip.setObjective(t - quicksum(j * v for j, v in enumerate(more_vars[20:-40])))
+
+    return scip
+
+
+def test_strong_branching():
+    scip = create_model()
+
+    strong_branch_rule = StrongBranchingRule(scip, idempotent=False)
+    scip.includeBranchrule(strong_branch_rule, "strong branch rule", "custom strong branching rule",
+                           priority=10000000, maxdepth=-1, maxbounddist=1)
+
+    scip.optimize()
```

Comment:
Are asserts not needed here? To verify that the objective is correct (we're not cutting off nodes incorrectly, for example).

Reply:
Hmmmmmmmmmmm. You would calculate the objective of the problem without this and then add an assert statement?

Reply:
I did that. Added an extra safety layer to the test!

Reply:
Yeah, but no need to do the optimization every time, just do it once, get the objective, and then assert it's equal to it